hypha/apply/todo/views.py

```diff
@@ -0,0 +1,138 @@
+from django.contrib.auth.models import Group
+from django.contrib.contenttypes.models import ContentType
+from django.db.models import Count
+
+from .models import Task
+from .options import get_task_template
+from .services import validate_user_groups_uniqueness, validate_user_uniquness
+
+
+def add_task_to_user(code, user, related_obj):
+    """
+    Add task for a user
+    input:
+        code: TASKS_CODE_CHOICES.keys()
+        user: User object
+        related_obj: Object - Submission, Project, Invoice, Report
+    output: task - Task object / None in case of no creation
+    """
+    user_uniqueness = validate_user_uniquness(
+        code=code, user=user, related_obj=related_obj
+    )
+    if user_uniqueness:
+        task = Task.objects.create(code=code, user=user, related_object=related_obj)
+        return task
+    return None
+
+
+def add_task_to_user_group(code, user_group, related_obj):
+    """
+    Add task for user_groups
+    input:
+        code: TASKS_CODE_CHOICES.keys()
+        user_group: Queryset - Group objects
+        related_obj: Object - Submission, Project, Invoice, Report
+    output: task - Task object / None in case of no creation
+    """
+    user_groups_uniqueness = validate_user_groups_uniqueness(
+        code=code, user_groups=user_group, related_obj=related_obj
+    )
+    if user_groups_uniqueness:
+        task = Task.objects.create(code=code, related_object=related_obj)
+        groups = [Group.objects.filter(id=group.id).first() for group in user_group]
+        task.user_group.add(*groups)
+        return task
+    return None
+
+
+def remove_tasks_for_user(code, user, related_obj):
+    """
+    Remove task for a user
+    input:
+        code: TASKS_CODE_CHOICES.keys()
+        user: User object
+        related_obj: Object - Submission, Project, Invoice, Report
+    output: None
+    """
+    task = Task.objects.filter(
+        code=code,
+        user=user,
+        related_content_type=ContentType.objects.get_for_model(related_obj).id,
+        related_object_id=related_obj.id,
+    ).first()
+    if task:
+        task.delete()
```

Comment:
```suggestion
    Task.objects.filter(
        code=code,
        user=user,
        related_content_type=ContentType.objects.get_for_model(related_obj).id,
        related_object_id=related_obj.id,
    ).delete()
```

Consider using a more optimized version of delete